commit 96c84703f1cf6ea43617f9565166681cd71df104
Merge: c0e75905caf3 951a6bf30667
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Tue Jan 21 16:09:47 2025 -0800

    Merge tag 'drm-next-2025-01-17' of https://gitlab.freedesktop.org/drm/kernel
    
    Pull drm updates from Dave Airlie:
     "There are two external interactions of note, the msm tree pull in some
      opp tree, hopefully the opp tree arrives from the same git tree
      however it normally does.
    
      There is also a new cgroup controller for device memory, that is used
      by drm, so is merging through my tree. This will hopefully help open
      up gpu cgroup usage a bit more and move us forward.
    
      There is a new accelerator driver for the AMD XDNA Ryzen AI NPUs.
    
      Then the usual xe/amdgpu/i915/msm leaders and lots of changes and
      refactors across the board:
    
      core:
       - device memory cgroup controller added
       - Remove driver date from drm_driver
       - Add drm_printer based hex dumper
       - drm memory stats docs update
       - scheduler documentation improvements
    
      new driver:
       - amdxdna - Ryzen AI NPU support
    
      connector:
       - add a mutex to protect ELD
       - make connector setup two-step
    
      panels:
       - Introduce backlight quirks infrastructure
       - New panels: KDB KD116N2130B12, Tianma TM070JDHG34-00,
       - Multi-Inno Technology MI1010Z1T-1CP11
    
      bridge:
       - ti-sn65dsi83: Add ti,lvds-vod-swing optional properties
       - Provide default implementation of atomic_check for HDMI bridges
       - it605: HDCP improvements, MCCS Support
    
      xe:
       - make OA buffer size configurable
       - GuC capture fixes
       - add ufence and g2h flushes
       - restore system memory GGTT mappings
       - ioctl fixes
       - SRIOV PF scheduling priority
       - allow fault injection
       - lots of improvements/refactors
       - Enable GuC's WA_DUAL_QUEUE for newer platforms
       - IRQ related fixes and improvements
    
      i915:
       - More accurate engine busyness metrics with GuC submission
       - Ensure partial BO segment offset never exceeds allowed max
       - Flush GuC CT receive tasklet during reset preparation
       - Some DG2 refactor to fix DG2 bugs when operating with certain CPUs
       - Fix DG1 power gate sequence
       - Enabling uncompressed 128b/132b UHBR SST
       - Handle hdmi connector init failures, and no HDMI/DP cases
       - More robust engine resets on Haswell and older
    
      i915/xe display:
       - HDCP fixes for Xe3Lpd
       - New GSC FW ARL-H/ARL-U
       - support 3 VDSC engines 12 slices
       - MBUS joining sanitisation
       - reconcile i915/xe display power mgmt
       - Xe3Lpd fixes
       - UHBR rates for Thunderbolt
    
      amdgpu:
       - DRM panic support
       - track BO memory stats at runtime
       - Fix max surface handling in DC
       - Cleaner shader support for gfx10.3 dGPUs
       - fix drm buddy trim handling
       - SDMA engine reset updates
       - Fix doorbell ttm cleanup
       - RAS updates
       - ISP updates
       - SDMA queue reset support
       - Rework DPM powergating interfaces
       - Documentation updates and cleanups
       - DCN 3.5 updates
       - Use a pm notifier to more gracefully handle VRAM eviction on
         suspend or hibernate
       - Add debugfs interfaces for forcing scheduling to specific engine
         instances
       - GG 9.5 updates
       - IH 4.4 updates
       - Make missing optional firmware less noisy
       - PSP 13.x updates
       - SMU 13.x updates
       - VCN 5.x updates
       - JPEG 5.x updates
       - GC 12.x updates
       - DC FAMS updates
    
      amdkfd:
       - GG 9.5 updates
       - Logging improvements
       - Shader debugger fixes
       - Trap handler cleanup
       - Cleanup includes
       - Eviction fence wq fix
    
      msm:
       - MDSS:
          - properly described UBWC registers
          - added SM6150 (aka QCS615) support
       - DPU:
          - added SM6150 (aka QCS615) support
          - enabled wide planes if virtual planes are enabled (by using two
            SSPPs for a single plane)
          - added CWB hardware blocks support
       - DSI:
          - added SM6150 (aka QCS615) support
       - GPU:
          - Print GMU core fw version
          - GMU bandwidth voting for a740 and a750
          - Expose uche trap base via uapi
          - UAPI error reporting
    
      rcar-du:
       - Add r8a779h0 Support
    
      ivpu:
       - Fix qemu crash when using passthrough
    
      nouveau:
       - expose GSP-RM logging buffers via debugfs
    
      panfrost:
       - Add MT8188 Mali-G57 MC3 support
    
      rockchip:
       - Gamma LUT support
    
      hisilicon:
       - new HIBMC support
    
      virtio-gpu:
       - convert to helpers
       - add prime support for scanout buffers
    
      v3d:
       - Add DRM_IOCTL_V3D_PERFMON_SET_GLOBAL
    
      vc4:
       - Add support for BCM2712
    
      vkms:
       - line-per-line compositing algorithm to improve performance
    
      zynqmp:
       - Add DP audio support
    
      mediatek:
       - dp: Add sdp path reset
       - dp: Support flexible length of DP calibration data
    
      etnaviv:
       - add fdinfo memory support
       - add explicit reset handling"
    
    * tag 'drm-next-2025-01-17' of https://gitlab.freedesktop.org/drm/kernel: (1070 commits)
      drm/bridge: fix documentation for the hdmi_audio_prepare() callback
      doc/cgroup: Fix title underline length
      drm/doc: Include new drm-compute documentation
      cgroup/dmem: Fix parameters documentation
      cgroup/dmem: Select PAGE_COUNTER
      kernel/cgroup: Remove the unused variable climit
      drm/display: hdmi: Do not read EDID on disconnected connectors
      drm/tests: hdmi: Add connector disablement test
      drm/connector: hdmi: Do atomic check when necessary
      drm/amd/display: 3.2.316
      drm/amd/display: avoid reset DTBCLK at clock init
      drm/amd/display: improve dpia pre-train
      drm/amd/display: Apply DML21 Patches
      drm/amd/display: Use HW lock mgr for PSR1
      drm/amd/display: Revised for Replay Pseudo vblank control
      drm/amd/display: Add a new flag for replay low hz
      drm/amd/display: Remove unused read_ono_state function from Hwss module
      drm/amd/display: Do not elevate mem_type change to full update
      drm/amd/display: Do not wait for PSR disable on vbl enable
      drm/amd/display: Remove unnecessary eDP power down
      ...

diff --cc drivers/gpu/drm/display/drm_bridge_connector.c
index 3a5a4f92c979,32108307de66..56f977bbe62d
--- a/drivers/gpu/drm/display/drm_bridge_connector.c
+++ b/drivers/gpu/drm/display/drm_bridge_connector.c
@@@ -460,9 -596,8 +596,11 @@@ struct drm_connector *drm_bridge_connec
  		return ERR_PTR(-EINVAL);
  
  	if (bridge_connector->bridge_hdmi) {
 +		if (!connector->ycbcr_420_allowed)
 +			supported_formats &= ~BIT(HDMI_COLORSPACE_YUV420);
 +
+ 		bridge = bridge_connector->bridge_hdmi;
+ 
  		ret = drmm_connector_hdmi_init(drm, connector,
  					       bridge_connector->bridge_hdmi->vendor,
  					       bridge_connector->bridge_hdmi->product,